src/librustc_mir/monomorphize/collector.rs

```diff
@@ -758,15 +758,15 @@ fn visit_instance_use<'tcx>(
         | ty::InstanceDef::Item(..)
         | ty::InstanceDef::FnPtrShim(..)
         | ty::InstanceDef::CloneShim(..) => {
-            output.push(create_fn_mono_item(instance, source));
+            output.push(create_fn_mono_item(tcx, instance, source));
         }
     }
 }
 
 // Returns `true` if we should codegen an instance in the local crate.
 // Returns `false` if we can just link to the upstream crate and therefore don't
 // need a mono item.
-fn should_monomorphize_locally<'tcx>(tcx: TyCtxt<'tcx>, instance: &Instance<'tcx>) -> bool {
+fn should_codegen_locally<'tcx>(tcx: TyCtxt<'tcx>, instance: &Instance<'tcx>) -> bool {
```

Comment:
Probably too much for this PR but maybe open an issue about replacing "mono item" terminology?

Simplest thing would probably be "codegen item" but we can bikeshed "item" out as well.

Reply:
I've not filed an issue regarding this yet.

Reply:
Filed #74432.